From c176904c9efb173c9bc368124ebffcab19b7eac8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian@centricular.com>
Date: Thu, 12 Jan 2017 15:55:52 +0200
Subject: [PATCH] opencv: Fix build with OpenCV < 3

We only need the opencv2/bgsegm.hpp header for OpenCV >= 3 and have
compat code for older versions.
---
 configure.ac | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index cea3e7ec6..7444c5749 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2730,8 +2730,13 @@ AG_GST_CHECK_FEATURE(OPENCV, [opencv plugins], opencv, [
                       opencv2/imgproc/imgproc_c.h \
                       opencv2/objdetect/objdetect.hpp \
                       opencv2/video/background_segm.hpp], [], [something_not_found=yes])
-    dnl check opencv_contrib headers (not always present in opencv distributions)
-    AC_CHECK_HEADERS([opencv2/bgsegm.hpp], [], [something_not_found=yes])
+
+    dnl Only required for OpenCV >= 3, we have compat code for older versions
+    PKG_CHECK_MODULES(OPENCV_3, opencv >= 3, [
+      dnl check opencv_contrib headers (not always present in opencv distributions)
+      AC_CHECK_HEADERS([opencv2/bgsegm.hpp], [], [something_not_found=yes])],
+      [NOT_A_PROBLEM=yes])
+
     CPPFLAGS=$OLD_CPPFLAGS
     AC_LANG([C])
 
-- 
2.11.0

